framework/auth/views.py

```diff
@@ -185,6 +190,33 @@ def auth_logout(redirect_url=None):
     return resp
 
 
+def auth_email_logout(token, user):
+    """When a user is adding an email or merging an account, add the email to the user and log them out.
+    """
+    redirect_url = web_url_for('auth_login') + '?existing_user={}'.format(urllib.quote_plus(user.email))
+    try:
+        user.verify_token(token)
+    except KeyError:
+        raise HTTPError(http.BAD_REQUEST, data={
+            'message_short': "Bad token",
+            'message_long': "There was a key error with the token"
+        })
+    except ExpiredTokenError:
+        status.push_status_message('The private link you used is expired.')
+    try:
+        user_merge = User.find_one(Q('emails', 'eq', user.email_verifications[token]['email'].lower()))
```

Comment:
As suggested in a previous comment, you can get rid of `verify_token` in favor of using `get_unconfrimed_email_for_token`. That way you can avoid accessing `user.email_verifications` directly here.